sonar-java-plugin/src/main/java/org/sonar/plugins/java/JavaSonarWayProfile.java

```diff
@@ -30,40 +38,51 @@
 import org.sonar.java.checks.CheckList;
 import org.sonarsource.api.sonarlint.SonarLintSide;
 
-import java.io.IOException;
-import java.net.URL;
-import java.nio.charset.StandardCharsets;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-
 /**
  * Replacement for org.sonar.plugins.squid.SonarWayProfile
  */
 @SonarLintSide
 public class JavaSonarWayProfile extends ProfileDefinition {
 
   private final RuleFinder ruleFinder;
-  public JavaSonarWayProfile(RuleFinder ruleFinder) {
+  private final Settings settings;
+
+  public JavaSonarWayProfile(RuleFinder ruleFinder, Settings settings) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "RuleFinder"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "Settings"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)